class-tgm-plugin-activation.php

```diff
@@ -439,7 +439,7 @@ public function install_plugins_page() {
 				<?php $plugin_table->prepare_items(); ?>
 
 				<?php
-				if ( isset( $this->message ) ) {
+				if ( isset( $this->message ) && is_string( $this->message ) && '' !== $this->message ) {
```

Comment:
I'll change it to your first option - the second is definitely incorrect as that would pass `new stdClass`, `(int) 5` etc